pkg/util/cloudproviders/azure/azure.go

```diff
@@ -174,6 +174,30 @@ func getHostnameWithConfig(ctx context.Context, config model.Config) (string, er
 	return name, nil
 }
 
+var hostCCRIDFetcher = cachedfetch.Fetcher{
+	Name: "Azure Host CCRID",
+	Attempt: func(ctx context.Context) (interface{}, error) {
+		rg, err := getResponse(ctx,
+			metadataURL+"/metadata/instance/compute/resourceId?api-version=2021-02-01&format=text")
+		if err != nil {
+			return "", fmt.Errorf("unable to query metadata endpoint: %s", err)
+		}
+		return rg, nil
+	},
+}
+
+// GetHostCCRID returns the Canonical Cloud Resource ID for the Azure host
+func GetHostCCRID(ctx context.Context) (string, error) {
```

Comment:
We should add a test for this new function

Reply:
Tests being added here: https://github.com/DataDog/datadog-agent/pull/39636